Revert commit 99137b78 (first seen in 4.20.0), which breaks dhcp over gre6.

See https://bugzilla.kernel.org/show_bug.cgi?id=202147

--- a/net/packet/af_packet.c
+++ b/net/packet/af_packet.c
@@ -2627,8 +2627,6 @@
 		proto	= saddr->sll_protocol;
 		addr	= saddr->sll_halen ? saddr->sll_addr : NULL;
 		dev = dev_get_by_index(sock_net(&po->sk), saddr->sll_ifindex);
-		if (addr && dev && saddr->sll_halen < dev->addr_len)
-			goto out_put;
 	}
 
 	err = -ENXIO;
@@ -2827,8 +2825,6 @@
 		proto	= saddr->sll_protocol;
 		addr	= saddr->sll_halen ? saddr->sll_addr : NULL;
 		dev = dev_get_by_index(sock_net(sk), saddr->sll_ifindex);
-		if (addr && dev && saddr->sll_halen < dev->addr_len)
-			goto out_unlock;
 	}
 
 	err = -ENXIO;
